### PR TITLE
Add delete taskrun command

### PR DESCRIPTION
**Pull request by 16yuki0702**
# Changes
Add deleting the taskrun feature for #193.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
you can delete taskrun from tkn.
```

### CODE REVIEW

**Comment by chmouel** [comment]
same comment as https://github.com/tektoncd/cli/pull/236#issuecomment-525622712 @16yuki0702 any chances we can have the same as #235, defaulting to ask if we really want to delete unless -f ?

**Comment by chmouel** [comment]
The CI error seems unreleated to your code, i have added an issue here https://github.com/tektoncd/pipeline/issues/1252

**Comment by 16yuki0702** [comment]
> The CI error seems unreleated to your code, i have added an issue here tektoncd/pipeline#1252

I got it, thanks!!

File: docs/cmd/tkn_taskrun_delete.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+### Synopsis
+
+Delete a taskrun in a namespace
+
+### Examples
+
+
+# Delete a TaskRun of name 'foo' in namespace 'bar'
```
Comment (gavinfish) [inline docs/cmd/tkn_taskrun_delete.md]: I think there is no need to have a H1 for `Delete a TaskRun of name 'foo' in namespace 'bar'`.

File: docs/cmd/tkn_taskrun_delete.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+tkn taskrun delete foo -n bar
+
+tkn tr rm foo -n bar",
```
Comment (gavinfish) [inline docs/cmd/tkn_taskrun_delete.md]: Why there is a single `"`?

File: pkg/cmd/taskrun/delete.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+func deleteTaskRun(s *cli.Stream, p cli.Params, trName string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return fmt.Errorf("Failed to create tekton client")
+	}
+
+	if err := cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).Delete(trName, &metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("Failed to delete taskrun %q: %s", trName, err)
```
Comment (gavinfish) [inline pkg/cmd/taskrun/delete.go]: Error string should not be capitalized.

File: pkg/cmd/taskrun/delete.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	}
+	f.AddFlags(c)
+	c.Flags().BoolVarP(&opts.forceDelete, "force", "f", false, "Whether to force deletion (default: false)")
+	return c
+}
+
+func deleteTaskRun(s *cli.Stream, p cli.Params, trName string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return fmt.Errorf("Failed to create tekton client")
```
Comment (gavinfish) [inline pkg/cmd/taskrun/delete.go]: Error string should not be capitalized.

File: pkg/cmd/taskrun/delete.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	fmt.Fprintf(s.Out, "Make sure you really want to delete taskrun %q (y/n): ", trName)
+	scanner := bufio.NewScanner(s.In)
+	for scanner.Scan() {
+		t := strings.TrimSpace(scanner.Text())
+		if t == "y" {
+			break
+		} else if t == "n" {
+			return fmt.Errorf("Canceled deleting taskrun %q", trName)
```
Comment (gavinfish) [inline pkg/cmd/taskrun/delete.go]: Error string should not be capitalized.

File: docs/cmd/tkn_taskrun_delete.md (hunk 1 above)
Comment (16yuki0702) [reply docs/cmd/tkn_taskrun_delete.md]: Thank you for your review!!
another docs also have a H1 these. https://github.com/tektoncd/cli/blob/master/docs/cmd/tkn_resource_describe.md
I think we need make a rule about document format.
(after that, we need correct these!)

File: docs/cmd/tkn_taskrun_delete.md (hunk 2 above)
Comment (16yuki0702) [reply docs/cmd/tkn_taskrun_delete.md]: oh I didn't realize it, thanks!!
I should correct it together with other documents, so how about making another PR after this??

File: pkg/cmd/taskrun/delete.go (hunk 3 above)
Comment (16yuki0702) [reply pkg/cmd/taskrun/delete.go]: Hmm, I found that there are a lot of Capitalized error (I run `git grep fmt.Errorf`)
I think It depends on project rule (if we really need to make it).
how about ??

Comment (gavinfish) [reply pkg/cmd/taskrun/delete.go]: Details at https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: docs/cmd/tkn_taskrun_delete.md (hunk 1 above)
Comment (gavinfish) [reply docs/cmd/tkn_taskrun_delete.md]: Agree.

File: docs/cmd/tkn_taskrun_delete.md (hunk 2 above)
Comment (gavinfish) [reply docs/cmd/tkn_taskrun_delete.md]: Cool.

File: pkg/cmd/taskrun/delete.go (hunk 3 above)
Comment (16yuki0702) [reply pkg/cmd/taskrun/delete.go]: oh I didn't know it, thank you very much!!
I should correct it at least my PR :thumbsup:

Comment (16yuki0702) [reply pkg/cmd/taskrun/delete.go]: Fix it !! thanks :thumbsup:

Comment (chmouel) [reply pkg/cmd/taskrun/delete.go]: Interesting that golangci doesnt pick this up,

**Pull request merged**